pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -135,11 +136,11 @@ type repository struct {
 	distribution.Repository
 
 	ctx              context.Context
-	limitClient      kcoreclient.LimitRangesGetter
 	registryOSClient client.Interface
 	registryAddr     string
 	namespace        string
 	name             string
+	config           *configuration.Configuration
```

Comment:
this is @legionus code for metrics... maybe we should just pass a boolean indicating if metrics are enabled.